core/controllers/platform_feature_test.py

```diff
@@ -78,17 +80,24 @@ def setUp(self):
             ]
         )
 
-        feature_services.ALL_FEATURES_LIST = param_names
+        # Here we use MyPy ignore because the expected type of ALL_FEATURES_LIST
+        # is a list of 'PARAM_NAMES' Enum, but here for testing purposes we are
+        # providing a list of str, which causes MyPy to throw an 'Incompatible
+        # types in assignment' error. Thus to avoid the error, we used ignore
+        # here.
+        feature_services.ALL_FEATURES_LIST = param_names  # type: ignore[assignment]
```

Comment:
While this was the code before, is this actually logically correct? The MyPy error here may be pointing out a real inconsistency in the test that could (and probably should) be corrected rather than ignoring the type error.

Reply:
I think this is not a problem because, in the same setup method, we are creating platform parameters with `ParamNames.PARAMETER_A` https://github.com/oppia/oppia/blob/f8a8d680c6f95dfc3c63a003484177ac06e4ce21/core/controllers/platform_feature_test.py#L56
and the value of `ParamNames.PARAMETER_A` is `parameter_a`.

and also, we are not changing it permanently we are reverting this mock behavior in teardown method by assigning the original values https://github.com/oppia/oppia/blob/f8a8d680c6f95dfc3c63a003484177ac06e4ce21/core/controllers/platform_feature_test.py#L87

Please let me know If i missed or not answered something from your question!

Reply:
Did some changes, PTAL!. Basically, now we are not assigning list of str, we are assigning list of ParamEnums.

Reply:
I think this is closer to what seems like a correct solution, but potentially introduces other confusion. Left a follow-up, but I think it's fine to resolve this thread now.